devicetypes/smartthings/zigbee-non-holdable-button.src/zigbee-non-holdable-button.groovy

```diff
@@ -0,0 +1,216 @@
+                }
+                
+            } else if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002) {
+                def zs = new ZoneStatus(zigbee.convertToInt(descMap.value, 16))
+                map = translateZoneStatus(zs)
+            } else if (descMap?.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == zigbee.ATTRIBUTE_IAS_ZONE_STATUS && descMap?.value) {
+                map = translateZoneStatus(new ZoneStatus(zigbee.convertToInt(descMap?.value)))
+            }
+        }
+    }
+
+    log.debug "Parse returned $map"
+    def result = map ? createEvent(map) : [:]
+
+    if (description?.startsWith('enroll request')) {
+        List cmds = zigbee.enrollResponse()
+        log.debug "enroll response: ${cmds}"
+        result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+    }
+    return result
+}
+
+private Map parseIasMessage(String description) {
+    ZoneStatus zs = zigbee.parseZoneStatus(description)
+
+    translateZoneStatus(zs)
+}
+
+private Map translateZoneStatus(ZoneStatus zs) {
+	if (zs.isAlarm1Set()) {
+        return getButtonResult('pushed')
+    }
+}
+
+private Map getBatteryResult(rawValue) {
+    log.debug "Battery rawValue = ${rawValue}"
+    def linkText = getLinkText(device)
+
+    def result = [:]
+
+    def volts = rawValue / 10
+
+    if (!(rawValue == 0 || rawValue == 255)) {
+        result.name = 'battery'
+        result.translatable = true
+        result.descriptionText = "${ device.displayName } battery was ${ value }%"
+        if (device.getDataValue("manufacturer") == "SmartThings") {
+            volts = rawValue // For the batteryMap to work the key needs to be an int
+            def batteryMap = [28: 100, 27: 100, 26: 100, 25: 90, 24: 90, 23: 70,
+                              22: 70, 21: 50, 20: 50, 19: 30, 18: 30, 17: 15, 16: 1, 15: 0]
+            def minVolts = 15
+            def maxVolts = 28
+
+            if (volts < minVolts)
+                volts = minVolts
+            else if (volts > maxVolts)
+                volts = maxVolts
+            def pct = batteryMap[volts]
+            result.value = pct
+        } else {
+            def minVolts = 2.1
+            def maxVolts = 3.0
+            def pct = (volts - minVolts) / (maxVolts - minVolts)
+            def roundedPct = Math.round(pct * 100)
+            if (roundedPct <= 0)
+                roundedPct = 1
+            result.value = Math.min(100, roundedPct)
+        }
+
+    }
+
+    return result
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+    log.debug "Battery Percentage rawValue = ${rawValue} -> ${rawValue / 2}%"
+    def result = [:]
+
+    if (0 <= rawValue && rawValue <= 200) {
+        result.name = 'battery'
+        result.translatable = true
+        result.descriptionText = "{{ device.displayName }} battery was {{ value }}%"
+        result.value = Math.round(rawValue / 2)
+    }
+
+    return result
+}
+
+private Map getButtonResult(value) {
```

Comment:
Since this DTH only supports `pushed` you can simplify it by removing the handling for held and double tap